en/services/mavlink_id_assignment.md

```diff
@@ -0,0 +1,52 @@
+
+In a single system you may also have multiple components of the same type.
+In this case you will need to ensure that each has a unique component id.
+
+The following section suggests a process to allow MAVLink components to automatically update themselves with the correct ids.
+
+## System Id Assignment
+
+First change the system ID of the autopilot component.
+This is usually done with a flight-stack specific parameter such as [MAV_SYS_ID](https://docs.px4.io/main/en/advanced_config/parameter_reference.html#MAV_SYS_ID) (PX4) or [SYSID_THISMAV](https://ardupilot.org/copter/docs/parameters.html#sysid-thismav-mavlink-system-id-of-this-vehicle) (ArduCopter).
+
+> **Note** [MAV_CMD_DO_SET_SYS_CMP_ID](../messages/development.md#MAV_CMD_DO_SET_SYS_CMP_ID) is a work-in-progress command for setting and changing system and command ids.
+
+Then reboot the vehicle off-network, so that the only autopilot visible to components should be the one on the vehicle (you can disconnect your telemetry radio, for example).
+
+On boot, components should look for autopilot `HEARTBEAT` messages.
+
+If HEARTBEAT messages are detected from:
+
+- Only one autopilot within 3 seconds of the first autopilot heartbeat being detected, the component should set its system id to match that of the autopilot.
+- Multiple autopilots within the first 3 seconds then the autopilot should keep its current system id.
+  Note that this is true even if autopilot heartbeats from corresponding autopilot are not seen.
+- No autopilots then components should keep their current system id.
+
+This ensures that components will latch to an unambiguous system id if available, and otherwise keep their current system id.
+
+## Component Id Assignment
+
+Component IDs should be manually assigned by the system integrator.
+
+In practise most components are factory-shipped using the first "allocated" component ID for their type from the [MAV_COMPONENT](../messages/common.md#MAV_COMPONENT) enum, such as [MAV_COMP_ID_CAMERA](../messages/common.html#MAV_COMP_ID_CAMERA) for cameras, and [MAV_COMP_ID_GIMBAL](../messages/common.md#MAV_COMP_ID_GIMBAL) for gimbals.
+If the vehicle has only one component of a particular type, then this is usually sufficient to avoid clashes.
+
+When there are multiple instances of a particular component, each will need a unique id.
+The easiest way to do this is to sequentially allocate component IDs from the allocated ranges, such as [MAV_COMP_ID_CAMERA2](../messages/common.md#MAV_COMP_ID_CAMERA2).
+The mechanisms to set and change component ids are component-dependent (and some components may not allow the component id to be changed).
+
+Note that in theory any component can have any id.
+However associating components with their well-known component ids makes it easier to, for example, associate a gimbal with a camera, or capture an image using a particular camera.
```

Comment:
@julianoes @auturgy I modified this to make it generic about both sysid and compid assignment.

My thinking is that component ID assignment should be done manually if needed, if only because it only needs to be done once, but also because it is useful to use the default values for knowing what your cameras are. If they changed ID on boot or whatever you would be stuffed.

Reply:
I would agrees components need to be assigned manually.